modules/avahi: Enable IPv6 by default

frederictobiasc opened the pull request:
Avahi's default for `use-ipv6` is yes as well. I see no reason why we should do this differently.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fpletz commented:
@frederictobiasc I agree that these options should probably also be turned on for a better user experience. Avahi is also being used to announce and discover the presence of services on the network via dbus. For instance, pipewire/pulseaudio is able to detect other instances/sinks over the network and make them available automatically. The nss module is not needed for this to work.

fpletz commented:
We also have to add release notes if we change the defaults.

frederictobiasc commented:
>  The nss module is not needed for this to work.

@fpletz But do you support enabling nss modules by default as well? I'll update the PR accordingly.

fpletz commented:
I just discovered #258424 where IPv6 was explicitly disabled. This doesn't make sense and IMHO shouldn't have been done on that PR.

@SuperSandro2000 Since it wasn't mentioned in that PR, do you recall the reasoning behind disabling IPv6?

> But do you support enabling nss modules by default as well? I'll update the PR accordingly.

I would prefer another PR since IPv6 support itself should be less controversial.

frederictobiasc commented:
> We also have to add release notes if we change the defaults.

Done!

> I would prefer another PR since IPv6 support itself should be less controversial.

I opened https://github.com/NixOS/nixpkgs/pull/361191 for that.

SuperSandro2000 commented:
> I would prefer another PR since IPv6 support itself should be less controversial.

It is super controversial and should stay turned off.

> Since it wasn't mentioned in that PR, do you recall the reasoning behind disabling IPv6?

https://github.com/NixOS/nixpkgs/commit/bba808dbfa89c8d974b248f4bf3c7716312dc704

https://github.com/NixOS/nixpkgs/pull/258424

https://github.com/avahi/nss-mdns#:~:text=in%20such%20a%20situation%20causes%20long%20timeouts%20when%20resolving%20hosts

fpletz commented:
@SuperSandro2000 But as far as I understand the problem is NSS and not IPv6 support per se in Avahi, right? That's why I was advocating to split NSS into a different PR. IPv6 support shouldn't trigger the problem you encountered when only `nssmdns4` is enabled as far as I understand it - please correct me if I'm wrong here. Note we are not enabling NSS here.

I would personally love to have IPv6 support enabled for all applications we ship by default.

frederictobiasc commented:
Ok so it's fine to merge this now and continue the discussion regarding nss in https://github.com/NixOS/nixpkgs/pull/361191, correct?

frederictobiasc committed:
modules/avahi: Enable IPv6 by default

Avahi's default for `use-ipv6` is yes as well. I see no reason why we
should do this differently.

bjornfor commented:
> IMO the whole point of enabling avahi is being able to resolve *.local domains and being resolveable as well. The currently required configuration to do that is big:
> 
> ```
> services.avahi = {
>   enable = true;
>   nssmdns6 = true;
>   nssmdns4 = true;
>   publish.addresses = true;
>   publish.enable = true;
>   ipv6 = true;
> };
> ```
> 
> On other distributions like arch this just works™ Does anyone see any issue with making that the default apart from `avahi.enable`?

NixOS < 16.03 only required `services.avahi = { enable = true; nssmdns = true; };` (not sure about ipv6, but at least the rest). The `...publish = ` settings were added in NixOS 16.03. Maybe you'll find some reasoning behind the smaller and smaller feature set of `services.avahi.enable = true` in git log?

bjornfor commented on nixos/doc/manual/release-notes/rl-2505.section.md:
```diff
@@ -77,6 +77,8 @@
 
 - Cinnamon has been updated to 6.4.
 
+- `services.avahi.ipv6` now defaults to true.
```
How does it interact with `networking.enableIPv6`? Maybe the option should default to `networking.enableIPv6`?
